datasync/chaindatafetcher/kafka/config.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2020 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+package kafka
+
+import "github.com/Shopify/sarama"
+
+type KafkaConfig struct {
+	saramaConfig *sarama.Config // kafka client configurations.
+	brokers      []string       // brokers is a list of broker URLs.
+	partitions   int32          // partitions is the number of partitions of a topic.
```

Comment:
To make sure what I understand is correct, partition is a unit of distributed processing and replica is a unit of replication, right?

Reply:
So, a topic is kinda message queue which consists of several partitions to process message in parallel. The replica(= replication factor) is how many partitions are copied to other brokers.

For example, Broker1 is a leader for topic1-partition1 and the other brokers have the replicated t1-p1. 

![image](https://user-images.githubusercontent.com/45347815/94102617-7ff6ea00-fe6d-11ea-9647-98d66f920331.png)

Reply:
Fully Understood :)